federation/pkg/federation-controller/service/service_controller.go

```diff
@@ -614,6 +614,23 @@ func (sc *ServiceController) reconcileService(serviceKey types.NamespacedName) (
 			return statusError, err
 		}
 
+		// Copy ingresses of unreachable clusters for comparison and do not delete those ingresses
```

Comment:
I would prefer to just ignore the unreadyClusters (i.e. don't delete their DNS records), rather than explicitly handle them, if possible.  I need to review the other code to determine whether that's better or worse than what's being done here.